source/blog/2018-06-15-release-notes.md

```diff
@@ -0,0 +1,209 @@
+
+```
+  import com.twitter.finagle.{Filter, Service}
+  import com.twitter.finagle.service.ConstantService
+  import com.twitter.util.Future
+
+  ...
+
+  val svc: Service[Int, Int] = filters.andThen(new ConstantService[Int, Int](Future.value(2)))
+```
+
+Then, `svc.toString` would thus return the String:
+"com.foo.MyFilter1.andThen(com.foo.MyFilter2).andThen(com.twitter.finagle.service.ConstantService(ConstFuture(2)))"
+
+Filter implementations are permitted to override their `toString` implementations which would
+replace the default of `getClass.getName`. [25474da1](https://github.com/twitter/finagle/commit/25474da16ff5cbaf18a764f199e42e569c152452)
+
+  * finagle-core: Make `Filter.TypeAgnostic` an abstract class for Java usability.
+    [6534e459](https://github.com/twitter/finagle/commit/6534e459302f48ba252cd7729eb57653c3b49b93)
+
+  * finagle-core: `c.t.f.filter.NackAdmissionFilter` is now public. [1855566d](https://github.com/twitter/finagle/commit/1855566d418f11d6159266a0ee767b12f454acab)
+
+  * finagle-core: Extended `c.t.f.ssl.KeyCredentials` and `c.t.f.ssl.TrustCredentials` to work
+    with `javax.net.ssl.KeyManagerFactory` and `javax.net.ssl.TrustManagerFactory` respectively.
+    [c863ca0b](https://github.com/twitter/finagle/commit/c863ca0b606455d78350588595b3190165efe26b)
+
+Breaking API Changes:
+
+  * finagle-core: Rename `DeadlineFilter.Param(maxRejectFraction)` to
+    `DeadlineFilter.MaxRejectFraction(maxRejectFraction)` to reduce confusion
+    when adding additional params.
+    [cb6975af](https://github.com/twitter/finagle/commit/cb6975af67c76d1fc917b68d5eba8939fd0a85bc)
+
+Bug Fixes:
+
+  * finagle-http2: `StreamTransportFactory` now marks itself as dead/closed when it runs out of
+    HTTP/2 stream IDs instead of stalling. This allows the connection to be closed/reestablished in
+    accordance with the spec [c5554673](https://github.com/twitter/finagle/commit/c555467350eb80c6894199c05942178625540a86)
+
+  * finagle-netty4: `SslServerSessionVerifier` is now supplied with the proper peer address
+    rather than `Address.failing`. [263e9786](https://github.com/twitter/finagle/commit/263e97866386d8d82f89838b045248daadad1564)
+
+  * finagle-thrift/thriftmux: Disabled client side per-endpoint stats by default for client
+    ServicePerEndpoint. It can be set via `c.t.f.thrift.RichClientParam` or a `with`-method
+    as `Thrift{Mux}.client.withPerEndpointStats`. [0f1ff3eb](https://github.com/twitter/finagle/commit/0f1ff3eb88de7e270254956b8bf5f2936c7d9947)
+
+  * finagle-netty4: Avoid NoClassDefFoundError if netty-transport-native-epoll is not available
+    on the classpath.
+
+
+### [Finatra](https://github.com/twitter/finatra/) ###
+
+Added:
+
+* finatra: Add HTTP route, Thrift method, and Filter information to the Library 
+  registry. ``PHAB_ID=D177583``
+
+* finatra-inject/inject-logback: Add an `c.t.inject.logback.AsyncAppender` to
+  provide metrics about the underlying queue. ``PHAB_ID=D173278``
```

Comment:
Don't we usually link to the commits?

Reply:
something must have borked in my process of running the phab linker.